node/src/command.rs

```diff
@@ -309,21 +310,45 @@ pub fn run() -> Result<()> {
 
 			Ok(())
 		},
-		Some(Subcommand::Benchmark(cmd)) =>
-			if cfg!(feature = "runtime-benchmarks") {
-				let runner = cli.create_runner(cmd)?;
-				let chain_spec = &runner.config().chain_spec;
-				with_runtime_or_err!(chain_spec, {
-					{
-						runner.sync_run(|config| cmd.run::<Block, Executor>(config))
+		Some(Subcommand::Benchmark(cmd)) => {
+			let runner = cli.create_runner(cmd)?;
+			let chain_spec = &runner.config().chain_spec;
+			// Switch on the concrete benchmark sub-command-
+			with_runtime_or_err!(chain_spec, {
+				{
+					match cmd {
+						BenchmarkCmd::Pallet(cmd) =>
+							if cfg!(feature = "runtime-benchmarks") {
+								runner.sync_run(|config| {
+									cmd.run::<Block, Executor>(config)
+								})
+							} else {
+								Err("Benchmarking wasn't enabled when building the node. \
+					You can enable it with `--features runtime-benchmarks`."
+									.into())
+							},
+						BenchmarkCmd::Block(cmd) => runner.sync_run(|config| {
+							let partials = service::new_partial::<RuntimeApi, Executor, _>(
+								&config,
+								crate::service::parachain_build_import_queue,
+							)?;
+							cmd.run(partials.client)
+						}),
+						BenchmarkCmd::Storage(cmd) => runner.sync_run(|config| {
+							let partials = service::new_partial::<RuntimeApi, Executor, _>(
+								&config,
+								crate::service::parachain_build_import_queue,
+							)?;
+							let db = partials.backend.expose_db();
+							let storage = partials.backend.expose_storage();
+
+							cmd.run(config, partials.client.clone(), db, storage)
+						}),
+						BenchmarkCmd::Overhead(_) => Err("Unsupported benchmarking command".into()),
```

Comment:
what is block, storage and overhead? (well, overhead does nothing, but what about the other two?)

Reply:
I am not sure honestly. The documentation has not been updated yet